src/rgw/rgw_rest_swift.cc

```diff
@@ -503,6 +505,33 @@ static void get_rmattrs_from_headers(const req_state * const s,
   }
 }
 
+static int get_swift_versioning_settings(
+  req_state * const s,
+  boost::optional<std::string>& swift_ver_location)
+{
+  /* Removing the Swift's versions location has lower priority than setting
+   * a new one. That's the reason why we're handling it first. */
+  const std::string vlocdel =
+    s->info.env->get("HTTP_X_REMOVE_VERSIONS_LOCATION", "");
+  if (vlocdel.size()) {
+    swift_ver_location = boost::in_place(std::string());
```

Comment:
`boost::optional` doesn't have such method. Moreover, at the moment it's empty and we need to construct `std::string` to fill it.